src/main/java/com/vmware/vipclient/i18n/messages/service/LocaleService.java

```diff
@@ -59,8 +61,12 @@ public LocaleService() {
         return respMap;
     }
 
-    public Map<String, String> getDisplayNamesFromCLDR(String language) {
-        Map<String, String> dispMap = null;
+    public Map<String, String> getDisplayNamesFromCLDR(String language, 
```

Comment:
Suggest substituting loop for recursion. That maybe more efficient.

Reply:
Efficiency wise, it is the same: O(n), n = number of items in msgSourceQueueIter.
It is exactly the same number of lines of code also, so I do not prefer anything. 
Nonetheless, I changed it to while loop (iteration).